components/bms_worker/src/Environment.c

```diff
@@ -0,0 +1,174 @@
+
+
+/******************************************************************************
+ *                             T Y P E D E F S
+ ******************************************************************************/
+
+typedef enum
+{
+    INIT = 0x00,
+    MEASURING,
+    DONE
+} Sensor_State_E;
+
+/******************************************************************************
+ *                               M A C R O S
+ ******************************************************************************/
+
+/******************************************************************************
+ *                           P U B L I C  V A R S
+ ******************************************************************************/
+
+/**
+ * @brief  Stores public struct for BMS Module
+ */
+Environment_S ENV;
+
+/******************************************************************************
+ *                         P R I V A T E  V A R S
+ ******************************************************************************/
+
+static Sensor_State_E ltc_state;
+static Sensor_State_E hs_state;
+
+/******************************************************************************
+ *            P U B L I C  F U N C T I O N  P R O T O T Y P E S
+ ******************************************************************************/
+
+/******************************************************************************
+ *          P R I V A T E  F U N C T I O N  P R O T O T Y P E S
+ ******************************************************************************/
+
+/******************************************************************************
+ *                       P U B L I C  F U N C T I O N S
+ ******************************************************************************/
+
+/**
+ * @brief  Environment Module Init function
+ */
+static void Environment_Init()
+{
+    ENV.state = ENV_INIT;
+
+    if (!LTC_Init())
+        ENV.state = ENV_ERROR;
+    if (!HS4011_Init())
+        ENV.state = ENV_ERROR;
+
+    if (ENV.state != ENV_ERROR)
+        ENV.state = ENV_RUNNING;
+}
+
+/**
+ * @brief  10Hz Environment periodic function
+ */
+static void Environment10Hz_PRD()
+{
+    if (ltc_state == MEASURING)
+    {
+        if (LTC_GetMeasurement())
+        {
+            ltc_state = DONE;
+
+            uint16_t tmp_min = UINT16_MAX;
+            uint16_t tmp_max = 0x00;
+            uint32_t tmp_avg = 0x00;
+
+            for (uint8_t i = 0; i < CHANNEL_COUNT; i++)
+            {
+                ENV.values.cells.cell_temps[i] = ltc_chip.temps[i];
+                if (ltc_chip.temps[i] < tmp_min) tmp_min = ltc_chip.temps[i];
+                if (ltc_chip.temps[i] > tmp_max) tmp_max = ltc_chip.temps[i];
+                tmp_avg += ltc_chip.temps[i];
+            }
+            
+            tmp_avg /= CHANNEL_COUNT;
+
+            ENV.values.cells.avg_temp = (uint16_t)tmp_avg;
+            ENV.values.cells.max_temp = tmp_max;
+            ENV.values.cells.min_temp = tmp_min;
```

Comment:
I think it would be a lot more clear if these quantities were floats. Also easier to work with elsewhere

Reply:
I try to keep it in stdint. I have no literature but as i understand soft floats are computationally heavy